app/controllers/spree/user_sessions_controller.rb

```diff
@@ -48,5 +50,11 @@ def redirect_back_or_default(default)
       redirect_to(session["spree_user_return_to"] || default)
       session["spree_user_return_to"] = nil
     end
+
+    def ensure_valid_locale
+      return unless spree_current_user && !available_locale?(spree_current_user.locale)
+
+      spree_current_user.update!(locale: I18n.default_locale)
```

Comment:
I dont think we need something specific for this controller?
I think we should just do this in this controller like we do in all other controllers:
before_action :set_locale

Reply:
The thing with this controller is that `spree_current_user` only exists _after_ the controller action has finished, as it's handling login. Before and during the action there is no logged in user, and in some cases there is _still_ no logged in user after the action has finished (for example if the given the username or password were incorrect). So the parts that persist the locale on the user's account in `I18nHelper#set_locale` do not work in this specific action, in this specific controller...

Reply:
ok, what about
after_action :set:_locale, only: :create
?

and then make set_locale handle it? I think we need to keep this logic in one place only.
If we really need to do something different here I think we need to extract set_locale to a service and call that service from here too, even if a different method than set_locale.

Reply:
So we do need to _set_ the locale _before_ the page loads, but in this case we also have to wait until _after_ the action to be able to persist a valid locale...

Reply:
:+1: